OpenInFirefoxClient/OpenInFirefoxControllerObjC.m

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
```

Comment:
So if this code came from Google (along with most of the other code in this file, I imagine), we need to keep the license from the original file. Where exactly did you copy this from?